internal/checker/flow.go

```diff
@@ -2442,15 +2442,17 @@ func (c *Checker) getFlowTypeInStaticBlocks(symbol *ast.Symbol, staticBlocks []*
 }
 
 func (c *Checker) isReachableFlowNode(flow *ast.FlowNode) bool {
-	result := c.isReachableFlowNodeWorker(flow, false /*noCacheCheck*/)
+	result := c.isReachableFlowNodeWorker(flow, false /*noCacheCheck*/, nil)
 	c.lastFlowNode = flow
 	c.lastFlowNodeReachable = result
 	return result
 }
 
-func (c *Checker) isReachableFlowNodeWorker(flow *ast.FlowNode, noCacheCheck bool) bool {
+func (c *Checker) isReachableFlowNodeWorker(flow *ast.FlowNode, noCacheCheck bool, reduceLabels []*ast.FlowReduceLabelData) bool {
 	var reduceLabelsBuffer [4]*ast.FlowReduceLabelData
-	reduceLabels := reduceLabelsBuffer[:0]
+	if reduceLabels == nil {
+		reduceLabels = reduceLabelsBuffer[:0]
+	}
```

Comment:
Now that this is passed down, I'm not entirely certain that this stack allocation optimization works.